### PR TITLE
[FLINK-12330][python]Add integrated Tox for ensuring compatibility of multi-version of python.

### PR DESCRIPTION
**Pull request by sunjincheng121**
## What is the purpose of the change
Add integrated Tox for ensuring compatibility of multi-version of python.

## Brief change log

  - Add lint-python.sh for integrated Conda and Tox.
  - Add the usage of how to run the lint-python.sh in README.md.

## Verifying this change
This change without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: flink-python/README.md [hunk 1]
```diff
@@ -10,15 +10,28 @@ In this initial version only Table API is supported, you can find the documentat
...
-Then enter the directory where this README.md file is located and execute `python setup.py install` to install PyFlink on your device.
+Currently, we can install PyFlink from Flink source code. Enter the directory where this README.md file is located and install PyFlink on your device by executing 
 
-## Running Tests
+```
+python setup.py install
+```
 
-Currently you can perform an end-to-end test of PyFlink in the directory where this file is located with the following command:
+## Running env verify example
```
Comment (dianfu) [inline flink-python/README.md]: This section seems unnecessary to me. I'm not sure in which case should we check the env manually. When users run ./dev/lint-python.sh, the env will be checked anyway.
Besides, I think that adding one specific test such as test_env_verify_example.py to check the env is not necessary as lint-python.sh will fail if the env is not set correctly.

File: flink-python/dev/lint-python.sh [hunk 2]
```diff
@@ -0,0 +1,508 @@
...
+
+function install_miniconda() {
+    OS_TO_CONDA_URL=("https://repo.continuum.io/miniconda/Miniconda3-latest-MacOSX-x86_64.sh" \
+        "https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh")
+    print_function "STEP" "download miniconda..."
+    if [ ! -f "$CONDA_INSTALL" ]; then
+        download ${OS_TO_CONDA_URL[$1]} $CONDA_INSTALL_SH
+        chmod +x $CONDA_INSTALL_SH
+        if [ $? -ne 0 ]; then
+            echo "Please mannually chmod +x $CONDA_INSTALL_SH"
```
Comment (dianfu) [inline flink-python/dev/lint-python.sh]: typo: mannually -> manually

File: flink-python/dev/lint-python.sh [hunk 3]
```diff
@@ -0,0 +1,508 @@
...
+    FLINK_IDENT_STRING="$USER"
+fi
+if [ "$HOSTNAME" == "" ]; then
+    HOSTNAME="$HOST"
+fi
+
+# the log file stores the checking result.
+LOG_FILE=$LOG_DIR/flink-$FLINK_IDENT_STRING-python-$HOSTNAME.log
+create_dir $LOG_DIR
+if [ $? -ne 0 ]; then
```
Comment (dianfu) [inline flink-python/dev/lint-python.sh]: create_dir will make the process exit if create file failed. So the check here seems unnecessary. What do you think?

File: flink-python/dev/lint-python.sh [hunk 4]
```diff
@@ -0,0 +1,508 @@
...
+            echo "conda install ${py_env[i]} failed.\
+            You can retry to exec the script."
+            exit 1
+        fi
+        print_function "STEP" "install python${py_env[i]}... [SUCCESS]"
+    done
+}
+
+# Install tox.
+# In some situations,you need to run the script with "sudo". e.g. sudo ./lint-python
```
Comment (WeiZhong94) [inline flink-python/dev/lint-python.sh]: execute "sudo ./lint-python" will print this message on my device: "sudo: ./lint-python: command not found".
Is it mean "sudo ./lint-python.sh" ?

File: flink-python/dev/lint-python.sh [hunk 5]
```diff
@@ -0,0 +1,508 @@
...
+# Stage 2:tox check python compatibility
+if [ $STAGE == "tox" ]; then
+    tox_check
+fi
+# Stage 3:code style test
+if [ $STAGE == "flake8" ]; then
+    flake8_check
+fi
+
+echo "All the checks are finished, the detailed information can be found in: $LOG_FILE"
```
Comment (WeiZhong94) [inline flink-python/dev/lint-python.sh]: IMO we should exit with a non-zero code when any checks failed just like we did in FLINK_HOME/tools/travis_controller.sh. The reason is that Travis-CI determines whether the script is executed successfully according to whether the exit code is 0 or not.

File: flink-python/dev/lint-python.sh (hunk 4 above)
Comment (sunjincheng121) [reply flink-python/dev/lint-python.sh]: Yes, `sudo ./lint-python.sh`.

File: flink-python/README.md (hunk 1 above)
Comment (sunjincheng121) [reply flink-python/README.md]: Make sense to me. +1

File: flink-python/dev/lint-python.sh (hunk 5 above)
Comment (sunjincheng121) [reply flink-python/dev/lint-python.sh]: Yes, I think we should quit immediately when we encounter an error.

**Comment by sunjincheng121** [comment]
Thanks for your review @dianfu @WeiZhong94 
I have updated the PR.
Please have another look! :)
Best, Jincheng

**Comment by dianfu** [comment]
LGTM. +1 to merge.

**Comment by WeiZhong94** [comment]
LGTM. +1 to merge.

**Comment by sunjincheng121** [comment]
Thanks @dianfu @WeiZhong94 !
Merging...
@flinkbot approve all

**Pull request closed without merging**